api/scaler/resource_provider.go

```diff
@@ -0,0 +1,47 @@
+package scaler
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	bytesize "github.com/zpatrick/go-bytesize"
+)
+
+type ResourceProvider struct {
+	AgentIsConnected bool              `json:"agent_connected"`
+	AvailableCPU     int               `json:"available_cpu"`
```

Comment:
[nit] I know this would change the logic, but should we change these to Used instead of available for consistency?

Reply:
Consistency with what? Also, by inverting this tracking, it would make calculating `HasResourcesFor` and the `ResourceProvider` model more complex, I think, since we'd have to keep track of the maximum amount allowed to be `Used`.

The current logic is `return consumer.CPU <= provider.AvailableCPU && consumer.Memory <= provider.AvailableMemory`.

I think it would have to be changed to something like `return consumer.CPU <= provider.MaxCPU - provider.UsedCPU && consumer.Memory <= provider.MaxMemory - provider.UsedMemory`, which seems excessive to me.

Reply:
✅